### DISCUSSION TITLE
requests to api.web3modal.org got cors error

### DISCUSSION THREAD

**Discussion by jaan1025** [WalletConnect/walletconnect-monorepo#4626] [Developer Support] [answered]
<img width="821" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/22076610/46f27f18-6c11-4b26-894e-e7d6f7cb4995">

<img width="640" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/22076610/a61ded6b-1801-44c9-b1b6-6cc6c5d299ba">

<img width="735" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/22076610/a85fe40d-e057-4a1e-917a-1730b25a871f">

The projectId is provided, and I also add my domain into "Allowed Domains" in cloud.

Any more configurations needed?

- "@web3modal/wagmi": "^5.0.1"
- "wagmi": "^2.10.2"

**Comment by glitch-txs** [comment]
Hi, make sure to follow this
https://docs.walletconnect.com/cloud/relay#allowlist

**Reply by jaan1025** [reply]
Yes, I'm sure I've configured Allowlist correctly. Could this possibly be related to the area I'm in?

**Reply by jaan1025** [reply]
And I found that I cannot select an origin like this. It shows "No options" here.
<img width="505" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/22076610/1fec8930-ed67-4d06-96fd-5e1689c86ceb">

**Reply by glitch-txs** [reply]
Could you try with a new project ID with an empty Allowlist? If it works it means that the configuration is incorrect.

**Reply by jaan1025** [reply]
<img width="818" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/22076610/8d3dffc1-3751-456b-8830-062c76911c50">

still not working

**Reply by Cali93** [reply]
Hey @jaan1025,

Could you please share the response and also check your console for the exact error you're getting.
Are you using any extension that might mess up with CORS ?

**Reply by jaan1025** [reply]
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/22076610/f36d03c2-4f6d-4182-9eb6-aa971773dc4e)
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/22076610/ea6bd4d3-aef3-4a58-b63c-e3890da50718)

Failed to load response data.

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/22076610/5f65ceb2-ca7a-4fe3-a8b3-332cb76dcd2d)

However, sometimes the API call is successful. But when I click "All Wallets," it throws an error again.

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/22076610/3b1e302f-0586-4ede-b3ac-e3600bba224d)

And the "getAssetImage" api always failed.

<img width="823" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/22076610/791fa369-d051-49ce-b898-a17b6dda25fb">

And I'm not using extension to modify access-control- headers

**Reply by Cali93** [reply, marked as answer]
Hey @jaan1025, as you can see on the CORS error in your console, the request gets rejected because you're providing a `traceparent` header which is forbidden by our CORS policy.

**Reply by jaan1025** [reply]
Ah! I see. Sorry for not reading the error message carefully. This header came from my monitoring SDK. It worked fine after I excluded this domain in the configuration. Thank you very much.
